### PR TITLE
doc: update coding guidelines

### PR DESCRIPTION
**Pull request by shiqingg**
This patch updates some rules in coding guidelines.

Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Ok to verify

File: doc/developer-guides/coding_guidelines.rst [hunk 1]
```diff
@@ -3495,27 +3495,37 @@ LE-03: Use of type attribute 'packed' is allowed
...
+
+This feature refers to section 6.32.1 in GCC 7.3 Manual.
+
+LE-06: Use of builtin-type '__builtin_va_list' is allowed
 =========================================================
 
 This feature refers to section 6.20 in GCC 7.3 Manual.
 
-LE-05: Use of builtin-function '__builtin_va_arg' is allowed
+LE-07: Use of builtin-function '__builtin_va_arg' is allowed
```
Comment (dbkinder) [inline doc/developer-guides/coding_guidelines.rst]: Did you consider combining these into one guideline, for example,
```
Use of built-in functions '__builtin_va_arg`, `__builtin_va_start",  '__builtin_va_end', and '__builtin_offsetof' are allowed.
```

**Comment by shiqingg** [comment]
Hi @dbkinder, yes, we could combine the similar rules into one. 
I will re-write these rules.

**Comment by shiqingg** [comment]
Hi @dbkinder,
New version has been pushed.
Could you please help review it again?
Thanks!

**Pull request merged**